### PR TITLE
Add script `platform_tests/test_platform_info.py` into T0 PR checker.

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

In PR [#13220](https://github.com/sonic-net/sonic-mgmt/pull/13220), we add a batch of control plane test scripts into onboarding T0 PR checker, but some of them failed. In this PR, we fix the failed test script `platform_tests/test_platform_info.py` and let it run successfully in T0 PR checker. 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
In PR [#13220](https://github.com/sonic-net/sonic-mgmt/pull/13220), we add a batch of control plane test scripts into onboarding T0 PR checker, but some of them failed. In this PR, we fix the failed test script `platform_tests/test_platform_info.py` and let it run successfully in T0 PR checker. 

#### How did you verify/test it?
```
=========================== short test summary info ============================
SKIPPED [1] common/helpers/assertions.py:16: No PSU controller for vlab-01, skip rest of the testing in this case
SKIPPED [3] platform_tests/thermal_control_test_helper.py:123: No mocker defined for this platform x86_64-kvm_x86_64-r0
============== 2 passed, 4 skipped, 1 warning in 64.33s (0:01:04) ==============
```

### CODE REVIEW

File: tests/platform_tests/test_platform_info.py [hunk 1]
```diff
@@ -173,7 +173,9 @@ def get_healthy_psu_num(duthost):
     """
     PSUUTIL_CMD = "sudo psuutil status"
     healthy_psus = 0
-    psuutil_status_output = duthost.command(PSUUTIL_CMD)
+    psuutil_status_output = duthost.command(PSUUTIL_CMD, module_ignore_errors=True)
+    if psuutil_status_output['rc'] != 0:
```
Comment (ZhaohuiS) [inline tests/platform_tests/test_platform_info.py:177]: @yutongzhang-microsoft if rc !=0, is there any possibility that the command doesn't work or is something wrong on physical testbed? If so, how to handle this scenario?

File: tests/platform_tests/thermal_control_test_helper.py [hunk 2]
```diff
@@ -283,7 +283,11 @@ def restart_thermal_control_daemon(dut):
         'Restarting thermal control daemon on {}...'.format(dut.hostname))
     find_thermalctld_pid_cmd = 'docker exec -i pmon bash -c \'pgrep -f thermalctld\' | sort'
     output = dut.shell(find_thermalctld_pid_cmd)
-    assert output["rc"] == 0, "Run command '%s' failed" % find_thermalctld_pid_cmd
+
+    if dut.facts["asic_type"] == "vs" and output["rc"] == 0:
+        pytest.skip("Thermalctld doesn't support on vs testbed")
+    else:
+        assert output["rc"] == 0, "Run command '%s' failed" % find_thermalctld_pid_cmd
```
Comment (ZhaohuiS) [inline tests/platform_tests/thermal_control_test_helper.py:290]: @yutongzhang-microsoft if you prefer to use format() to concentrate string, do you mind keeping using a same format for whole script?

File: tests/platform_tests/test_platform_info.py (hunk 1 above)
Comment (yutongzhang-microsoft) [reply tests/platform_tests/test_platform_info.py:177]: I will add another condition here, `if duthost.facts["asic_type"] == "vs"`

File: tests/platform_tests/thermal_control_test_helper.py (hunk 2 above)
Comment (yutongzhang-microsoft) [reply tests/platform_tests/thermal_control_test_helper.py:290]: Sure, I will keep the same format in this script

**Review by wangxin — APPROVED** [review]

**Review by wangxin — CHANGES_REQUESTED** [review]
Maybe the better idea is to expect different rc and output when DUT is vs setup.
Use `pytest.skip` here does not make sense.
If there is no point running the platform related commands at all, then add device_type "vs" mark is a better idea.

**Comment by yutongzhang-microsoft** [comment]
> Maybe the better idea is to expect different rc and output when DUT is vs setup. Use `pytest.skip` here does not make sense. If there is no point running the platform related commands at all, then add device_type "vs" mark is a better idea.

Hi, Xin, some part of the test such as running a command is worth to be tested as we discussed offline. So I use return instead of skip, to execute to left part.

**Review by xwjiang-ms — APPROVED** [review]
LGTM

**Review by wangxin — APPROVED** [review]

**Comment by Gfrom2016** [comment]
@yutongzhang-microsoft Do we need to cherry-pick this to 202311 and 202405?

**Comment by yutongzhang-microsoft** [comment]
> @yutongzhang-microsoft Do we need to cherry-pick this to 202311 and 202405?

These changes are aimed at fitting vs testbed, will have no influence on physical. You can cherry pick if you need.

**Pull request merged**